File: T6/codigos2021/T06_ComunicacionFragments/app/src/main/java/com/borja/t06_comunicacionfragments/fragments/FragmentDos.java
package com.borja.t06_comunicacionfragments.fragments;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.borja.t06_comunicacionfragments.R;

public class FragmentDos extends Fragment {

    private TextView textView;
    private String texto;

    public static FragmentDos newInstance(String texto) {


        Bundle bundle = new Bundle();
        bundle.putString("texto", texto);
        FragmentDos fragmentDos = new FragmentDos();
        fragmentDos.setArguments(bundle);
        return fragmentDos;
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        texto = this.getArguments().getString("texto");
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_dos_layout, container, false);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        textView = getView().findViewById(R.id.texto_fragment_dos);
        // asocio texto con textView
        textView.setText(texto);
    }
}

File: Kotlin/T7/codigos2223/InicioFragments/app/src/main/java/com/example/iniciofragments/fragments/FragmentDos.kt
package com.example.iniciofragments.fragments

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.iniciofragments.databinding.FragmentDosBinding
import com.example.iniciofragments.databinding.FragmentHeaderBinding
import com.example.iniciofragments.databinding.FragmentUnoBinding

class FragmentDos: Fragment() {

    private lateinit var binding: FragmentDosBinding
    private lateinit var nombre: String

    // TODO 1. Creo el objeto que permite el paso de parametros
    companion object{
        fun newInstance(nombre: String): FragmentDos {

            val fragmentDos = FragmentDos()
            // TODO 2. Creo un bundle para poder argumentos
            val bundle = Bundle()
            bundle.putString("nombre",nombre)
            fragmentDos.arguments = bundle
            return fragmentDos
        }
    }

    // TODO 3. Recuperar argumentos
    override fun onAttach(context: Context) {
        super.onAttach(context)
        nombre = this.arguments?.getString("nombre") ?: "defecto"
    }



    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = FragmentDosBinding.inflate(inflater,container,false)
        return binding.root
    }

    override fun onStart() {
        super.onStart()
        binding.textoNombreF2.setText(nombre)
    }
}